packages/headless/src/test/mock-engine.ts

```diff
@@ -154,17 +158,25 @@ export interface MockInsightEngine
 export function buildMockInsightEngine(
   config: Partial<InsightEngine<InsightAppState>> = {}
 ): MockInsightEngine {
-  return buildMockCoreEngine(config, buildMockInsightState);
+  const engine = buildMockCoreEngine(config, buildMockInsightState, 'insight');
+  return {
+    ...engine,
+    executeFirstSearch: jest.fn(),
+  };
 }
 
 interface MockCoreEngine<T extends object> extends CoreEngine<T>, MockEngine {}
 
 function buildMockCoreEngine<T extends AppState>(
   config: Partial<CoreEngine<T>> = {},
-  mockState: () => T
+  mockState: () => T,
+  type: EngineType = 'search'
 ): MockCoreEngine<T> {
   const logger = pino({level: 'silent'});
-  const storeConfiguration = configureMockStore(logger);
+  const storeConfiguration =
+    type === 'search'
+      ? configureMockStore(logger)
+      : configureInsightMockStore(logger);
   const coreState = buildCoreState(config, mockState);
   const store = storeConfiguration(coreState);
```

Comment:
Why not expose a hook allowing to build the store instance instead? It could be something like that.

```typescript
function buildMockCoreEngine<T extends AppState>(
  config: Partial<CoreEngine<T>> = {},
  mockState: () => T,
  buildStore: (logger, state) => MockStoreCreator<AppState, DispatchExts> = (logger, state) => configureMockStore(logger)(state)
): MockCoreEngine<T> {
  const logger = pino({level: 'silent'});
  const coreState = buildCoreState(config, mockState);
  const store = buildStore(logger, coreState);
  ...
}
```

The code above is just to share the idea, I did not try it though.